packages/web-client/app/utils/web3-strategies/layer1-chain.ts

```diff
@@ -202,7 +202,16 @@ export default abstract class Layer1ChainWeb3Strategy
   }
 
   private updateWalletInfo(accounts: string[]) {
-    this.walletInfo = new WalletInfo(accounts);
+    let newWalletInfo = new WalletInfo(accounts);
+    if (this.walletInfo.isEqualTo(newWalletInfo)) {
```

Comment:
Before this condition, the `this.refreshBalances();` was always called (in case there are accounts). But now, it won't be called in case the `walletInfo` is equal due to the early return. I'm not sure but can this be problematic?

Reply:
You're right, this is a behaviour change. I think we aren't using `updateWalletInfo` to force balance refreshes at the moment, and we have a separate method to do it, so I would lean towards using an early return regardless.

Reply:
👍